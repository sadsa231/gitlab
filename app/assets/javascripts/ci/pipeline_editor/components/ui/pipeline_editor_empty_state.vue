<script>
import { GlButton, GlSprintf } from '@gitlab/ui';
import { __ } from '~/locale';
import PipelineEditorFileNav from '~/ci/pipeline_editor/components/file_nav/pipeline_editor_file_nav.vue';

export default {
  components: {
    GlButton,
    GlSprintf,
    PipelineEditorFileNav,
  },
  i18n: {
    title: __('Optimize your workflow with CI/CD Pipelines'),
    body: __(
      'Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
    btnText: __('Configure pipeline'),
    externalCiNote: __("This project's pipeline configuration is located outside this repository"),
    externalCiInstructions: __(
      'To edit the pipeline configuration, you must go to the project or external site that hosts the file.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
    usesExternalConfig: {
      default: false,
      type: Boolean,
      required: false,
    },
  },
  methods: {
    createEmptyConfigFile() {
      this.$emit('createEmptyConfigFile');
    },
  },
};
</script>
<template>
  <div>
    <pipeline-editor-file-nav v-on="$listeners" />
    <div class="gl-display-flex gl-flex-direction-column gl-align-items-center gl-mt-11">
      <img :src="emptyStateIllustrationPath" />
      <div
        v-if="usesExternalConfig"
        class="gl-display-flex gl-flex-direction-column gl-align-items-center"
      >
        <h1 class="gl-font-size-h1">{{ $options.i18n.externalCiNote }}</h1>
        <p class="gl-mt-3">{{ $options.i18n.externalCiInstructions }}</p>
      </div>
      <div v-else class="gl-display-flex gl-flex-direction-column gl-align-items-center">
        <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
        <p class="gl-mt-3">
          <gl-sprintf :message="$options.i18n.body">
            <template #code="{ content }">
              <code>{{ content }}</code>
            </template>
          </gl-sprintf>
        </p>
        <gl-button
          variant="confirm"
          class="gl-mt-3"
          data-qa-selector="create_new_ci_button"
          @click="createEmptyConfigFile"
        >
          {{ $options.i18n.btnText }}
        </gl-button>
      </div>
    </div>
  </div>
</template>
