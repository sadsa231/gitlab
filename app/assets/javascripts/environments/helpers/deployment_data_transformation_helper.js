import { getIdFromGraphQLId } from '~/graphql_shared/utils';

/**
 * This function transforms Commit object coming from GraphQL to object compatible with app/assets/javascripts/vue_shared/components/commit.vue author object
 * @param {Object} Commit
 * @returns {Object}
 */
export const getAuthorFromCommit = (commit) => {
  if (commit.author) {
    return {
      username: commit.author.name,
      path: commit.author.webUrl,
      avatar_url: commit.author.avatarUrl,
    };
  }
  return {
    username: commit.authorName,
    path: `mailto:${commit.authorEmail}`,
    avatar_url: commit.authorGravatar,
  };
};

/**
 * This function transforms deploymentNode object coming from GraphQL to object compatible with app/assets/javascripts/vue_shared/components/commit.vue
 * @param {Object} deploymentNode
 * @returns {Object}
 */
export const getCommitFromDeploymentNode = (deploymentNode) => {
  if (!deploymentNode.commit) {
    throw new Error("deploymentNode argument doesn't have 'commit' field", deploymentNode);
  }
  return {
    title: deploymentNode.commit.message,
    commitUrl: deploymentNode.commit.webUrl,
    shortSha: deploymentNode.commit.shortId,
    tag: deploymentNode.tag,
    commitRef: {
      name: deploymentNode.ref,
    },
    author: getAuthorFromCommit(deploymentNode.commit),
  };
};

export const convertJobToDeploymentAction = (job) => {
  return {
    name: job.name,
    playable: job.playable,
    scheduledAt: job.scheduledAt,
    playPath: `${job.webPath}/play`,
  };
};

export const getActionsFromDeploymentNode = (deploymentNode, lastDeploymentName) => {
  if (!deploymentNode || !lastDeploymentName) {
    return [];
  }

  return (
    deploymentNode.job?.deploymentPipeline?.jobs?.nodes
      ?.filter((deployment) => deployment.name !== lastDeploymentName)
      .map(convertJobToDeploymentAction) || []
  );
};

/**
 * This function transforms deploymentNode object coming from GraphQL to object compatible with app/assets/javascripts/environments/environment_details/page.vue table
 * @param {Object} deploymentNode
 * @returns {Object}
 */
export const convertToDeploymentTableRow = (deploymentNode, environment) => {
  const { lastDeployment } = environment;
  const commit = getCommitFromDeploymentNode(deploymentNode);
  return {
    status: deploymentNode.status.toLowerCase(),
    id: deploymentNode.iid,
    triggerer: deploymentNode.triggerer,
    commit,
    job: deploymentNode.job && {
      webPath: deploymentNode.job.webPath,
      label: `${deploymentNode.job.name} (#${getIdFromGraphQLId(deploymentNode.job.id)})`,
    },
    created: deploymentNode.createdAt || '',
    deployed: deploymentNode.finishedAt || '',
    actions: getActionsFromDeploymentNode(deploymentNode, lastDeployment?.job?.name),
  };
};
