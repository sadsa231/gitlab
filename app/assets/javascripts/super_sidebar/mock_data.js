import { s__ } from '~/locale';

export const context = {
  title: 'Typeahead.js',
  link: '/',
  avatar: 'https://gitlab.com/uploads/-/system/project/avatar/278964/project_avatar.png?width=32',
};

export const contextSwitcherItems = {
  yourWork: { title: s__('Navigation|Your work'), link: '/', icon: 'work' },
  recentProjects: [
    {
      // eslint-disable-next-line @gitlab/require-i18n-strings
      title: 'Orange',
      subtitle: 'tropical-tree',
      link: '/tropical-tree',
      avatar:
        'https://gitlab.com/uploads/-/system/project/avatar/4456656/pajamas-logo.png?width=64',
    },
    {
      // eslint-disable-next-line @gitlab/require-i18n-strings
      title: 'Lemon',
      subtitle: 'tropical-tree',
      link: '/tropical-tree',
      avatar: 'https://gitlab.com/uploads/-/system/project/avatar/7071551/GitLab_UI.png?width=64',
    },
    {
      // eslint-disable-next-line @gitlab/require-i18n-strings
      title: 'Coconut',
      subtitle: 'tropical-tree',
      link: '/tropical-tree',
      avatar:
        'https://gitlab.com/uploads/-/system/project/avatar/4149988/SVGs_project.png?width=64',
    },
  ],
  recentGroups: [
    {
      title: 'Developer Evangelism at GitLab',
      subtitle: 'tropical-tree',
      link: '/tropical-tree',
      avatar:
        'https://gitlab.com/uploads/-/system/group/avatar/10087220/rainbow_tanuki.jpg?width=64',
    },
    {
      title: 'security-products',
      subtitle: 'tropical-tree',
      link: '/tropical-tree',
      avatar:
        'https://gitlab.com/uploads/-/system/group/avatar/11932235/gitlab-icon-rgb.png?width=64',
    },
    {
      title: 'Tanuki-Workshops',
      subtitle: 'tropical-tree',
      link: '/tropical-tree',
      avatar:
        'https://gitlab.com/uploads/-/system/group/avatar/5085244/Screenshot_2019-04-29_at_16.13.07.png?width=64',
    },
  ],
};
