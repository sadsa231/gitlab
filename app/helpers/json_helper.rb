# frozen_string_literal: true

module JsonHelper
  # These two JSON helpers are short-form wrappers for the Gitlab::Json
  # class, which should be used in place of .to_json calls or calls to
  # the JSON class.
  def json_generate(*args)
    Gitlab::Json.generate(*args)
  end

  def json_parse(*args)
    Gitlab::Json.parse(*args)
  end
end
