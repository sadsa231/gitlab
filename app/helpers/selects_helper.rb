# frozen_string_literal: true

module SelectsHelper
  def project_select_tag(id, opts = {})
    opts[:class] = [*opts[:class], 'ajax-project-select'].join(' ')

    unless opts.delete(:scope) == :all
      if @group
        opts['data-group-id'] = @group.id
      end
    end

    with_feature_enabled_data_attribute =
      case opts.delete(:with_feature_enabled)
      when 'issues'         then 'data-with-issues-enabled'
      when 'merge_requests' then 'data-with-merge-requests-enabled'
      end

    opts[with_feature_enabled_data_attribute] = true

    hidden_field_tag(id, opts[:selected], opts)
  end

  def select2_tag(id, opts = {})
    klass_opts = [opts[:class]]
    klass_opts << 'multiselect' if opts[:multiple]

    opts[:class] = klass_opts.join(' ')
    value = opts[:selected] || ''

    hidden_field_tag(id, value, opts)
  end
end

SelectsHelper.prepend_mod_with('SelectsHelper')
