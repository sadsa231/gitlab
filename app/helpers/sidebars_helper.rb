# frozen_string_literal: true

module SidebarsHelper
  include Nav::NewDropdownHelper

  def sidebar_tracking_attributes_by_object(object)
    sidebar_attributes_for_object(object).fetch(:tracking_attrs, {})
  end

  def sidebar_qa_selector(object)
    sidebar_attributes_for_object(object).fetch(:sidebar_qa_selector, nil)
  end

  def scope_qa_menu_item(object)
    sidebar_attributes_for_object(object).fetch(:scope_qa_menu_item, nil)
  end

  def scope_avatar_classes(object)
    %w[avatar-container rect-avatar s32].tap do |klasses|
      klass = sidebar_attributes_for_object(object).fetch(:scope_avatar_class, nil)
      klasses << klass if klass
    end
  end

  def project_sidebar_context(project, user, current_ref, ref_type: nil)
    context_data = project_sidebar_context_data(project, user, current_ref, ref_type: ref_type)
    Sidebars::Projects::Context.new(**context_data)
  end

  def group_sidebar_context(group, user)
    context_data = group_sidebar_context_data(group, user)

    Sidebars::Groups::Context.new(**context_data)
  end

  def super_sidebar_context(user, group:, project:)
    {
      name: user.name,
      username: user.username,
      avatar_url: user.avatar_url,
      assigned_open_issues_count: user.assigned_open_issues_count,
      assigned_open_merge_requests_count: user.assigned_open_merge_requests_count,
      todos_pending_count: user.todos_pending_count,
      issues_dashboard_path: issues_dashboard_path(assignee_username: user.username),
      create_new_menu_groups: create_new_menu_groups(group: group, project: project),
      support_path: support_url,
      display_whats_new: display_whats_new?
    }
  end

  private

  def create_new_menu_groups(group:, project:)
    new_dropdown_sections = new_dropdown_view_model(group: group, project: project)[:menu_sections]
    show_headers = new_dropdown_sections.length > 1
    new_dropdown_sections.map do |section|
      {
        name: show_headers ? section[:title] : '',
        items: section[:menu_items].map do |item|
          {
            text: item[:title],
            href: item[:href]
          }
        end
      }
    end
  end

  def sidebar_attributes_for_object(object)
    case object
    when Project
      sidebar_project_attributes
    when Group
      sidebar_group_attributes
    when User
      sidebar_user_attributes
    else
      {}
    end
  end

  def sidebar_project_attributes
    {
      tracking_attrs: sidebar_project_tracking_attrs,
      sidebar_qa_selector: 'project_sidebar',
      scope_qa_menu_item: 'Project scope',
      scope_avatar_class: 'project_avatar'
    }
  end

  def sidebar_group_attributes
    {
      tracking_attrs: sidebar_group_tracking_attrs,
      sidebar_qa_selector: 'group_sidebar',
      scope_qa_menu_item: 'Group scope',
      scope_avatar_class: 'group_avatar'
    }
  end

  def sidebar_user_attributes
    {
      tracking_attrs: sidebar_user_profile_tracking_attrs
    }
  end

  def sidebar_project_tracking_attrs
    tracking_attrs('projects_side_navigation', 'render', 'projects_side_navigation')
  end

  def sidebar_group_tracking_attrs
    tracking_attrs('groups_side_navigation', 'render', 'groups_side_navigation')
  end

  def sidebar_user_profile_tracking_attrs
    tracking_attrs('user_side_navigation', 'render', 'user_side_navigation')
  end

  def project_sidebar_context_data(project, user, current_ref, ref_type: nil)
    {
      current_user: user,
      container: project,
      current_ref: current_ref,
      ref_type: ref_type,
      jira_issues_integration: project_jira_issues_integration?,
      can_view_pipeline_editor: can_view_pipeline_editor?(project),
      show_cluster_hint: show_gke_cluster_integration_callout?(project)
    }
  end

  def group_sidebar_context_data(group, user)
    {
      current_user: user,
      container: group
    }
  end
end

SidebarsHelper.prepend_mod_with('SidebarsHelper')
