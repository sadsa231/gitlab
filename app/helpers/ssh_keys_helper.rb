# frozen_string_literal: true

module SshKeysHelper
  def ssh_key_delete_modal_data(key, path)
    title = _('Delete Key')

    {
        path: path,
        method: 'delete',
        qa_selector: 'delete_ssh_key_button',
        title: title,
        aria_label: title,
        modal_attributes: {
            'data-qa-selector': 'ssh_key_delete_modal',
            title: _('Are you sure you want to delete this SSH key?'),
            message: _('This action cannot be undone, and will permanently delete the %{key} SSH key') % { key: key.title },
            okVariant: 'danger',
            okTitle: _('Delete')
        },
        toggle: 'tooltip',
        placement: 'top',
        container: 'body'
    }
  end

  def ssh_key_revoke_modal_data(key, path)
    title = _('Revoke Key')

    {
        path: path,
        method: 'delete',
        qa_selector: 'revoke_ssh_key_button',
        title: title,
        aria_label: title,
        modal_attributes: {
            'data-qa-selector': 'ssh_key_revoke_modal',
            title: _('Are you sure you want to revoke this SSH key?'),
            message: _('This action cannot be undone, and will permanently delete the %{key} SSH key. All commits signed using this SSH key will be marked as unverified.') % { key: key.title },
            okVariant: 'danger',
            okTitle: _('Revoke')
        },
        toggle: 'tooltip',
        placement: 'top',
        container: 'body'
    }
  end

  def ssh_key_allowed_algorithms
    allowed_algorithms = Gitlab::CurrentSettings.allowed_key_types.flat_map do |ssh_key_type_name|
      Gitlab::SSHPublicKey.supported_algorithms_for_name(ssh_key_type_name)
    end

    quoted_allowed_algorithms = allowed_algorithms.map { |name| "'#{name}'" }

    Gitlab::Utils.to_exclusive_sentence(quoted_allowed_algorithms)
  end
end
