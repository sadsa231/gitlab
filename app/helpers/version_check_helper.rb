# frozen_string_literal: true

module VersionCheckHelper
  include Gitlab::Utils::StrongMemoize

  SECURITY_ALERT_SEVERITY = 'danger'

  def show_version_check?
    return false unless Gitlab::CurrentSettings.version_check_enabled
    return false if User.single_user&.requires_usage_stats_consent?

    current_user&.can_read_all_resources?
  end

  def gitlab_version_check
    VersionCheck.new.response
  end
  strong_memoize_attr :gitlab_version_check

  def show_security_patch_upgrade_alert?
    return false unless Feature.enabled?(:critical_security_alert) && show_version_check? && gitlab_version_check

    gitlab_version_check['severity'] === SECURITY_ALERT_SEVERITY
  end

  def link_to_version
    if Gitlab.pre_release?
      commit_link = link_to(Gitlab.revision, source_host_url + namespace_project_commits_path(source_code_group, source_code_project, Gitlab.revision))
      [Gitlab::VERSION, content_tag(:small, commit_link)].join(' ').html_safe
    else
      link_to Gitlab::VERSION, source_host_url + namespace_project_tag_path(source_code_group, source_code_project, "v#{Gitlab::VERSION}")
    end
  end

  def source_host_url
    Gitlab::Saas.com_url
  end

  def source_code_group
    'gitlab-org'
  end

  def source_code_project
    'gitlab-foss'
  end
end

VersionCheckHelper.prepend_mod
