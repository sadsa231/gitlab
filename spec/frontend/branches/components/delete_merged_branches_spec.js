import { GlButton, GlModal, GlFormInput, GlSprintf } from '@gitlab/ui';
import { mount } from '@vue/test-utils';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import { stubComponent } from 'helpers/stub_component';
import waitForPromises from 'helpers/wait_for_promises';
import { createMockDirective, getBinding } from 'helpers/vue_mock_directive';
import DeleteMergedBranches, { i18n } from '~/branches/components/delete_merged_branches.vue';
import { formPath, propsDataMock } from '../mock_data';

jest.mock('~/lib/utils/csrf', () => ({ token: 'mock-csrf-token' }));

let wrapper;

const stubsData = {
  GlModal: stubComponent(GlModal, {
    template:
      '<div><slot name="modal-title"></slot><slot></slot><slot name="modal-footer"></slot></div>',
  }),
  GlButton,
  GlFormInput,
  GlSprintf,
};

const createComponent = (mountFn = shallowMountExtended, stubs = {}) => {
  wrapper = mountFn(DeleteMergedBranches, {
    propsData: {
      ...propsDataMock,
    },
    directives: {
      GlTooltip: createMockDirective(),
    },
    stubs,
  });
};

const findDeleteButton = () => wrapper.findComponent(GlButton);
const findModal = () => wrapper.findComponent(GlModal);
const findConfirmationButton = () =>
  wrapper.findByTestId('delete-merged-branches-confirmation-button');
const findCancelButton = () => wrapper.findByTestId('delete-merged-branches-cancel-button');
const findFormInput = () => wrapper.findComponent(GlFormInput);
const findForm = () => wrapper.find('form');
const submitFormSpy = () => jest.spyOn(wrapper.vm.$refs.form, 'submit');

describe('Delete merged branches component', () => {
  beforeEach(() => {
    createComponent();
  });

  describe('Delete merged branches button', () => {
    it('has correct attributes, text and tooltip', () => {
      expect(findDeleteButton().attributes()).toMatchObject({
        category: 'secondary',
        variant: 'danger',
      });

      expect(findDeleteButton().text()).toBe(i18n.deleteButtonText);
    });

    it('displays a tooltip', () => {
      const tooltip = getBinding(findDeleteButton().element, 'gl-tooltip');

      expect(tooltip).toBeDefined();
      expect(tooltip.value).toBe(wrapper.vm.buttonTooltipText);
    });

    it('opens modal when clicked', () => {
      createComponent(mount);
      jest.spyOn(wrapper.vm.$refs.modal, 'show');
      findDeleteButton().trigger('click');

      expect(wrapper.vm.$refs.modal.show).toHaveBeenCalled();
    });
  });

  describe('Delete merged branches confirmation modal', () => {
    beforeEach(() => {
      createComponent(shallowMountExtended, stubsData);
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it('renders correct modal title and text', () => {
      const modalText = findModal().text();
      expect(findModal().props('title')).toBe(i18n.modalTitle);
      expect(modalText).toContain(i18n.notVisibleBranchesWarning);
      expect(modalText).toContain(i18n.protectedBranchWarning);
    });

    it('renders confirm and cancel buttons with correct text', () => {
      expect(findConfirmationButton().text()).toContain(i18n.deleteButtonText);
      expect(findCancelButton().text()).toContain(i18n.cancelButtonText);
    });

    it('renders form with correct attributes and hiden inputs', () => {
      const form = findForm();
      expect(form.attributes()).toEqual({
        action: formPath,
        method: 'post',
      });
      expect(form.find('input[name="_method"]').attributes('value')).toBe('delete');
      expect(form.find('input[name="authenticity_token"]').attributes('value')).toBe(
        'mock-csrf-token',
      );
    });

    it('matches snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('has a disabled confirm button by default', () => {
      expect(findConfirmationButton().props('disabled')).toBe(true);
    });

    it('keeps disabled state when wrong input is provided', async () => {
      findFormInput().vm.$emit('input', 'hello');
      await waitForPromises();
      expect(findConfirmationButton().props('disabled')).toBe(true);
      findConfirmationButton().trigger('click');

      expect(submitFormSpy()).not.toHaveBeenCalled();
      findFormInput().trigger('keyup.enter');

      expect(submitFormSpy()).not.toHaveBeenCalled();
    });

    it('submits form when correct amount is provided and the confirm button is clicked', async () => {
      findFormInput().vm.$emit('input', 'delete');
      await waitForPromises();
      expect(findDeleteButton().props('disabled')).not.toBe(true);
      findConfirmationButton().trigger('click');
      expect(submitFormSpy()).toHaveBeenCalled();
    });

    it('calls hide on the modal when cancel button is clicked', () => {
      const closeModalSpy = jest.spyOn(wrapper.vm.$refs.modal, 'hide');
      findCancelButton().trigger('click');
      expect(closeModalSpy).toHaveBeenCalled();
    });
  });
});
