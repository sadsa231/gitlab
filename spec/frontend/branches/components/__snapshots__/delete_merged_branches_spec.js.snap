// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete merged branches component Delete merged branches confirmation modal matches snapshot 1`] = `
<div>
  <b-button-stub
    class="gl-mr-3 gl-button btn-danger-secondary"
    data-qa-selector="delete_merged_branches_button"
    size="md"
    tag="button"
    type="button"
    variant="danger"
  >
    <!---->
     
    <!---->
      
    <span
      class="gl-button-text"
    >
      Delete merged branches
  
    </span>
  </b-button-stub>
   
  <div>
    <form
      action="/namespace/project/-/merged_branches"
      method="post"
    >
      <p>
        You are about to 
        <strong>
          delete all branches
        </strong>
         that were merged into 
        <code>
          master
        </code>
        .
      </p>
       
      <p>
        
        This may include merged branches that are not visible on the current screen.
      
      </p>
       
      <p>
        
        A branch won't be deleted if it is protected or associated with an open merge request.
      
      </p>
       
      <p>
        This bulk action is 
        <strong>
          permanent and cannot be undone or recovered
        </strong>
        .
      </p>
       
      <p>
        Plese type the following to confirm: 
        <code>
          delete
        </code>
        . 
        <b-form-input-stub
          aria-labelledby="input-label"
          autocomplete="off"
          class="gl-form-input gl-mt-2 gl-form-input-sm"
          data-qa-selector="delete_merged_branches_input"
          debounce="0"
          formatter="[Function]"
          type="text"
          value=""
        />
      </p>
       
      <input
        name="_method"
        type="hidden"
        value="delete"
      />
       
      <input
        name="authenticity_token"
        type="hidden"
        value="mock-csrf-token"
      />
    </form>
    <div
      class="gl-display-flex gl-flex-direction-row gl-justify-content-end gl-flex-wrap gl-m-0 gl-mr-3"
    >
      <b-button-stub
        class="gl-button"
        data-testid="delete-merged-branches-cancel-button"
        size="md"
        tag="button"
        type="button"
        variant="default"
      >
        <!---->
         
        <!---->
          
        <span
          class="gl-button-text"
        >
          
          Cancel
        
        </span>
      </b-button-stub>
       
      <b-button-stub
        class="gl-button"
        data-qa-selector="delete_merged_branches_confirmation_button"
        data-testid="delete-merged-branches-confirmation-button"
        disabled="true"
        size="md"
        tag="button"
        type="button"
        variant="danger"
      >
        <!---->
         
        <!---->
          
        <span
          class="gl-button-text"
        >
          Delete merged branches
        </span>
      </b-button-stub>
    </div>
  </div>
</div>
`;
